tests/initcoder_test.go

```diff
@@ -0,0 +1,78 @@
+package tests
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/Fantom-foundation/go-opera/tests/contracts/initcode"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInitCodeSizeLimitAndMetered(t *testing.T) {
+	r := require.New(t)
+
+	net, err := StartIntegrationTestNet(t.TempDir())
+	r.NoError(err)
+	defer net.Stop()
+
+	// Deploy the invalid start contract.
+	contract, receipt, err := DeployContract(net, initcode.DeployInitcode)
+	r.NoError(err)
+	r.Equal(types.ReceiptStatusSuccessful, receipt.Status, "failed to deploy contract")
+
+	const gasForOneWord uint64 = 54977
+
+	t.Run("create transaction with enough gas for init code", func(t *testing.T) {
+		// we need to provide same gas to all transactions so that we can compare the cost.
+		cost1Word, _ := createAndCheckContractWithCodeLenAndGas(r, net, contract, 30, gasForOneWord+4)
+		cost2Words, _ := createAndCheckContractWithCodeLenAndGas(r, net, contract, 42, gasForOneWord+4)
+		cost3Words, _ := createAndCheckContractWithCodeLenAndGas(r, net, contract, 90, gasForOneWord+4)
+		if cost2Words-cost1Word != 2 {
+			t.Errorf("cost difference between 1 and 2 words should be 2, instead got %d", cost2Words-cost1Word)
+		}
+		if cost3Words-cost1Word != 4 {
+			t.Errorf("cost difference between 1 and 3 words should be 4, instead got %d", cost3Words-cost1Word)
+		}
+	})
+
+	t.Run("create transaction without enough gas for init code", func(t *testing.T) {
+		receipt = createContractWithCodeLenAndGas(r, net, contract, 42, gasForOneWord)
+		r.Equal(types.ReceiptStatusFailed, receipt.Status,
+			"unexpectedly succeeded to create contract without enough gas")
+	})
+
+	const MAX_INITCODE_SIZE = 49152
+	const MAX_INITCODE_COST = (MAX_INITCODE_SIZE / 32) * 2
+	const someCost = gasForOneWord + MAX_INITCODE_COST + 9_199
```

Comment:
If I understand correctly, this should not be `someCost` but `sufficientGas`

Reply:
where does 9199 come from?

Reply:
this number was part of the initialization cost of the init code size. it was been rework so that the numbers and intents are clearer.